3rdparty/boost-1.65.0.patch

```diff
@@ -16,3 +16,49 @@ index c0557de71..c06eaea45 100644
 -#     pragma message("Unknown compiler version - please run the configure tests and report the results")
 -#  endif
 -#endif
+From f48fd09d021db9a28bd7b8452c175897e1af4485 Mon Sep 17 00:00:00 2001
+From: Romain Geissler <romain.geissler@amadeus.com>
+Date: Thu, 1 Mar 2018 09:59:33 +0100
+Subject: [PATCH] Ignore gcc 8 warnings.
+
+---
+ boost/mpl/assert.hpp | 15 +++++++++++++--
+ 1 file changed, 13 insertions(+), 2 deletions(-)
+
+diff --git a/boost/mpl/assert.hpp b/boost/mpl/assert.hpp
+index 1af1b058..e41b583c 100644
+--- a/boost/mpl/assert.hpp
++++ b/boost/mpl/assert.hpp
+@@ -184,16 +184,27 @@ template< typename P > struct assert_arg_pred_not
+     typedef typename assert_arg_pred_impl<p>::type type;
+ };
+ 
++#if defined(BOOST_GCC) && BOOST_GCC >= 80000
++#define BOOST_MPL_IGNORE_PARENTHESES_WARNING
++#pragma GCC diagnostic push
++#pragma GCC diagnostic ignored "-Wparentheses"
++#endif
++
+ template< typename Pred >
+-failed ************ (Pred::************ 
++failed ************ (Pred::************
```

Comment:
I do not quite understand this change, these two lines are identical, right? So I guess we do not need to do this change?

Reply:
Hey @qianzhangxa ,

If you look closely the change removed a trailing space ` `.
It's a bit unrelated to the fix but this is exactly the commit which was merged upstream - https://github.com/boostorg/mpl/commit/3c1910797cf2a9e6f18a0a929bec95837accfb9e - so better to keep it as-is.

Cheers,

Reply:
Yeah, you are right, sorry for my bad.